### PR TITLE
Fix quarkus.kubernetes.host

### LINKED ISSUE #8257: Quarkus Kubernetes - Host is ignored
**Describe the bug**
When using the Kubernetes Extension and adding exposed=true and host=service.bla.com
the host entry is not added to the generated ingress part. Host is always ""

**To Reproduce**
1.  Create a small application
2. add Kubernetes extension
3. Add config properties and build by execute package
4. Check the configs

**Configuration**
```properties
 quarkus:
    kubernetes:
      group: bla
      replicas: 2
      imagePullPolicy: Always
      host: service.bla.com
      expose: true

``` 

**Environment (please complete the following information):**
 - Output of `java -version`: 11
 - Quarkus version or git rev: 1.3.1.Finial
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):  maven

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #8257

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java [hunk 1]
```diff
@@ -96,6 +97,12 @@ private static void handleExpose(Config config, Map<String, Object> unPrefixed)
                     .orElse(false);
             if (unprefixedExpose || prefixedExpose) {
                 unPrefixed.put(DEKORATE_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, true);
+                for (PlatformConfiguration platformConfiguration : platformConfigurations) {
+                    if (platformConfiguration.getConfigName().equals(generator)) {
+                        platformConfiguration.getHost().map(h -> unPrefixed.put(DEKORATE_PREFIX + generator + ".host", h));
```
Comment (gastaldi) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java]:
```suggestion
                        platformConfiguration.getHost().ifPresent(h -> unPrefixed.put(DEKORATE_PREFIX + generator + ".host", h));
```

Comment (gastaldi) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java]: Or the whole for/if/break block could be written using streams :)

Comment (gastaldi) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java]:
```java
Arrays.stream(platformConfigurations)
    .filter(f->f.getConfigName().equals(generator))
    .map(PlatformConfiguration::getHost)
    .findFirst()
    .ifPresent(h-> unPrefixed.put(DEKORATE_PREFIX + generator + ".host", h))
```

Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java]: Yes to first, no to the second :)

**Review by gastaldi — APPROVED** [review]
LGTM, just need to squash the commits :)

**Comment by geoand** [comment]
Commits squashed :sunglasses:

**Pull request merged**